src/platform/Linux/bluez/BluezAdvertisement.cpp

```diff
@@ -170,6 +143,53 @@ CHIP_ERROR BluezAdvertisement::Init(const BluezEndpoint & aEndpoint, ChipAdvType
     return err;
 }
 
+CHIP_ERROR BluezAdvertisement::SetIntervals(AdvertisingIntervals aAdvIntervals)
+{
+    VerifyOrReturnError(mpAdv != nullptr, CHIP_ERROR_UNINITIALIZED);
+    // If the advertisement is already running, BlueZ will update the intervals
+    // automatically. There is no need to stop and restart the advertisement.
+    bluez_leadvertisement1_set_min_interval(mpAdv, aAdvIntervals.first * 0.625);
+    bluez_leadvertisement1_set_max_interval(mpAdv, aAdvIntervals.second * 0.625);
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR BluezAdvertisement::SetupServiceData(bool aExtendedAnnouncement)
```

Comment:
a bool as method arguments are generally not super descriptive. Like `SetupServiceData(true)` sounds like 'do set it up' . Consider having some enum class or passing in the flags and then you can check them here.

Reply:
Done